c-dependencies/js-compute-runtime/builtins/backend.cpp

```diff
@@ -0,0 +1,857 @@
+    definition.ssl_max_version = tlsMaxVersion.toInt32();
+    backend_config_mask |= BACKEND_CONFIG_SSL_MAX_VERSION;
+  } else {
+    definition.ssl_max_version = 0;
+  }
+
+  std::optional<std::string> certificateHostname;
+  auto certificateHostnameSlot = JS::GetReservedSlot(backend, Backend::Slots::CertificateHostname);
+  if (!certificateHostnameSlot.isNullOrUndefined()) {
+    JS::RootedString certificateHostnameString(cx, certificateHostnameSlot.toString());
+    size_t certificateHostname_len;
+    JS::UniqueChars certificateHostnameChars =
+        encode(cx, certificateHostnameString, &certificateHostname_len);
+    certificateHostname = std::string(certificateHostnameChars.get(), certificateHostname_len);
+  }
+
+  std::optional<std::string> caCertificate;
+  auto caCertificateSlot = JS::GetReservedSlot(backend, Backend::Slots::CaCertificate);
+  if (!caCertificateSlot.isNullOrUndefined()) {
+    JS::RootedString caCertificateString(cx, caCertificateSlot.toString());
+    size_t caCertificate_len;
+    JS::UniqueChars caCertificateChars = encode(cx, caCertificateString, &caCertificate_len);
+    caCertificate = std::string(caCertificateChars.get(), caCertificate_len);
+  }
+
+  std::optional<std::string> ciphers;
+  auto ciphersSlot = JS::GetReservedSlot(backend, Backend::Slots::Ciphers);
+  if (!ciphersSlot.isNullOrUndefined()) {
+    JS::RootedString ciphersString(cx, ciphersSlot.toString());
+    size_t ciphers_len;
+    JS::UniqueChars ciphersChars = encode(cx, ciphersString, &ciphers_len);
+    ciphers = std::string(ciphersChars.get(), ciphers_len);
+  }
+
+  std::optional<std::string> sniHostname;
+  auto sniHostnameSlot = JS::GetReservedSlot(backend, Backend::Slots::SniHostname);
+  if (!sniHostnameSlot.isNullOrUndefined()) {
+    JS::RootedString sniHostnameString(cx, sniHostnameSlot.toString());
+    size_t sniHostname_len;
+    JS::UniqueChars sniHostnameChars = encode(cx, sniHostnameString, &sniHostname_len);
+    sniHostname = std::string(sniHostnameChars.get(), sniHostname_len);
+  }
+
+  auto useSslSlot = JS::GetReservedSlot(backend, Backend::Slots::UseSsl);
+  if (!useSslSlot.isNullOrUndefined() && useSslSlot.toBoolean()) {
+    backend_config_mask |= BACKEND_CONFIG_USE_SSL;
+  }
+
+  JS::RootedString name(cx, JS::GetReservedSlot(backend, Backend::Slots::Name).toString());
+  size_t name_len;
+  JS::UniqueChars nameChars = encode(cx, name, &name_len);
+  auto name_cstr = nameChars.get();
+
+  JS::RootedString target(cx, JS::GetReservedSlot(backend, Backend::Slots::Target).toString());
+  size_t target_len;
+  JS::UniqueChars targetChars = encode(cx, target, &target_len);
+  auto target_cstr = targetChars.get();
+  if (hostOverride.has_value()) {
+    backend_config_mask |= BACKEND_CONFIG_HOST_OVERRIDE;
+    definition.host_override = hostOverride.value().c_str();
+    definition.host_override_len = hostOverride.value().length();
+  } else {
+    definition.host_override = nullptr;
+    definition.host_override_len = 0;
```

Comment:
I think all the else cases that are zeroing out the fields of `definition` should be unnecessary after adding the `memset` at the top.

Reply:
good catch 👍